### PR TITLE
Added an Infiltrate function to lua

### PR DESCRIPTION
**Pull request by abcdefg30**

### CODE REVIEW

File: OpenRA.Mods.Common/Scripting/Properties/CaptureProperties.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	}
+
+	[ScriptPropertyGroup("Targeting")]
+	public class ExternalCaptureProperties : ScriptActorProperties, Requires<ExternalCapturesInfo>
+	{
+		public ExternalCaptureProperties(ScriptContext context, Actor self)
+			: base(context, self) { }
+
+		[Desc("Captures the target actor using external-capturing logic.")]
+		public void ExternalCapture(Actor target)
```
Comment (pchote) [inline OpenRA.Mods.Common/Scripting/Properties/CaptureProperties.cs]: It would be nicer if this was also named `Capture`, the same way that `Mobile` and `Aircraft` expose alternative `Move` script properties, and the different versions of `Attack`.

**Comment by abcdefg30** [comment]
Updated.

File: OpenRA.Mods.Common/OpenRA.Mods.Common.csproj [hunk 2]
```diff
@@ -198,6 +198,7 @@
     <Compile Include="Scripting\Global\FacingGlobal.cs" />
     <Compile Include="Scripting\Global\HSLColorGlobal.cs" />
     <Compile Include="Scripting\Global\UserInterfaceGlobal.cs" />
+    <Compile Include="Scripting\Properties\CaptureProperties.cs" />
```
Comment (Mailaender) [inline OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]: I suggest to move this into a separate pull request as it doesn't provide a test case yet and is independent from infiltrate.

File: OpenRA.Mods.Common/Scripting/Properties/CaptureProperties.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+using System.Linq;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Scripting;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Scripting
+{
+	[ScriptPropertyGroup("Targeting")]
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Scripting/Properties/CaptureProperties.cs]: Grouping capturing and infiltration under "Targeting" somehow rubs me the wrong way, but I don't have any better suggestions, either.

File: OpenRA.Mods.RA/Scripting/Properties/InfiltrateProperties.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Linq;
```
Comment (obrakmann) [inline OpenRA.Mods.RA/Scripting/Properties/InfiltrateProperties.cs]: I think this isn't needed.

**Comment by obrakmann** [comment]
Looks good to me otherwise. :+1:

File: OpenRA.Mods.RA/Scripting/Properties/InfiltrateProperties.cs (hunk 4 above)
Comment (abcdefg30) [inline OpenRA.Mods.RA/Scripting/Properties/InfiltrateProperties.cs]: Fixed.

File: OpenRA.Mods.Common/Scripting/Properties/CaptureProperties.cs (hunk 3 above)
Comment (Mailaender) [inline OpenRA.Mods.Common/Scripting/Properties/CaptureProperties.cs]: We call it special abilities in the tooltips so maybe `Ability`?

**Comment by penev92** [comment]
I have to agree with the naming nit above about "Targeting".

**Comment by abcdefg30** [comment]
Updated. Renamed `Targeting` to `Ability`.
> I suggest to move this into a separate pull request as it doesn't provide a test case yet and is independent from infiltrate.

Done: #8614

**Comment by Mailaender** [comment]
Works as promised. :white_check_mark:

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/cb06d675a075bb7fe43ed1aa65b7841257550944...1a1d0246e9d44e9d6b86daba4c1412e4ec3ed2f2)
